REGULUS-1075: Security Group rules are overwritten when deploying 2 Stacks within the same Resource Group name and Security Group value on Azure

alicia-td opened the pull request:
### Changes
* made network security group name unique by using uuid
* made key vault unique by using uuid
* add permissions to role policy + resources to read image versions (to pull dev image versions for testing)
* add `KeyVaultName` to arm templates outputs to use for workspace deployments
* update `SecurityGroup` output to `NetworkSecurityGroupId` in arm templates for clarity

### Testing
* With NLB:
<img width="1792" alt="Screenshot 2024-07-17 at 5 01 02 PM" src="https://github.com/user-attachments/assets/9b7b310b-a39e-4828-9885-6e9f5526199b">

* Without LB:
<img width="1792" alt="Screenshot 2024-07-17 at 5 42 55 PM" src="https://github.com/user-attachments/assets/5157b995-ad4e-4447-a8fe-87d1ab968b55">

* Same name + uuid network security groups
<img width="1207" alt="Screenshot 2024-07-17 at 2 07 11 PM" src="https://github.com/user-attachments/assets/12dcd40b-ec7e-4243-8dd6-d5b6eccbc086">

aopanis reviewed (approved):
lgtm, not sure if we need the unique string on keyvault since I think we only create them in ai-unlimited or all-in-one but it might be nice for testing if we do several deployments in the same RG

alicia-td commented:
> lgtm, not sure if we need the unique string on keyvault since I think we only create them in ai-unlimited or all-in-one but it might be nice for testing if we do several deployments in the same RG

I added the unique string on key vault since I ran into a key vault existing during a redeployment (even though I deleted the key vault but it has not been purged, so it ran into a conflict issue). I think key vaults need to be unique across subscriptions.
